src/App.tsx

```diff
@@ -1,9 +1,223 @@
-import './App.scss';
+import { useMemo, useState, useContext } from 'react';
+import { useNavigate } from 'react-router';
+import {
+  Navigate,
+  Route,
+  Routes,
+  useLocation,
+  useSearchParams,
+} from 'react-router-dom';
+import { Context } from './Context';
+import { Header } from './Header';
+import { Navigation } from './Navigation';
+import { Home } from './Home';
+import { Products } from './Products';
+import { ProductDetails } from './ProductDetails';
+import { Favourites } from './Favourites';
+import { Cart } from './Cart';
+import { Footer } from './Footer';
+import { ProductTypes } from './types/productTypes';
 
-const App = () => (
-  <div className="App">
-    <h1>React Phone Catalog</h1>
-  </div>
-);
+const App = () => {
+  const { setQuery } = useContext(Context);
+  const { pathname } = useLocation();
+  const history = useNavigate();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [filterType, setFilterType] = useState('');
+  const page = searchParams.get('page' || '');
+  const perPage = searchParams.get('perPage' || '');
+  const sort = searchParams.get('sort' || '');
+  const filterQuery = searchParams.get('query' || '');
+  const activeProduct = JSON.parse(
+    localStorage.getItem('product') as string,
```

Comment:
better to move all these query param names to some enum and reuse them everywhere
(the same i suggest making with localStorage keys)